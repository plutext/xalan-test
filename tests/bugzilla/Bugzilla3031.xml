<foo>
  <bar>1</bar>
  <bar>2</bar>
  <bar>3</bar>
  <bang>
    <bat>1</bat>
  </bang>
  <bang>
    <bat>3</bat>
  </bang>
</foo>