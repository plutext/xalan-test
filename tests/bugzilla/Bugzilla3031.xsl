<?xml version="1.0"?>
<xsl:transform version="1.0"
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform">

  <xsl:template match="/">
    <first worksCorrectly="YES">
      <xsl:for-each select="/foo/bar">
        <xsl:for-each select="/foo/bang[bat=current()]">
          <xsl:copy-of select="."/>
        </xsl:for-each>
      </xsl:for-each>
    </first>
    <second worksCorrectly="YES! (should do the same thing)">
      <xsl:for-each select="/foo/bar">
        <xsl:for-each select="/foo/bang[bat[.=current()]]">
          <xsl:copy-of select="."/>
        </xsl:for-each>
      </xsl:for-each>
    </second>
  </xsl:template>

</xsl:transform>
