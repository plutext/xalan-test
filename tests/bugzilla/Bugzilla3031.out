<?xml version="1.0" encoding="UTF-8"?>
<first worksCorrectly="YES"><bang>
    <bat>1</bat>
  </bang><bang>
    <bat>3</bat>
  </bang></first><second worksCorrectly="YES! (should do the same thing)"><bang>
    <bat>1</bat>
  </bang><bang>
    <bat>3</bat>
  </bang></second>